plugins/PostProcessingPlugin/scripts/AnnealingOrDrying.py

```diff
@@ -0,0 +1,513 @@
+"""
+
+from UM.Application import Application
+from ..Script import Script
+from UM.Message import Message
+
+class AnnealingOrDrying(Script):
+
+    def initialize(self) -> None:
+        super().initialize()
+        # Get the Bed Temperature from Cura
+        bed_temp = str(Application.getInstance().getGlobalContainerStack().getProperty("material_bed_temperature", "value"))
+        self._instance.setProperty("startout_temp", "value", bed_temp)
+        # Get the Build Volume temperature if there is one
+        heated_build_volume = bool(Application.getInstance().getGlobalContainerStack().getProperty("machine_heated_build_volume", "value"))
+        curaApp = Application.getInstance().getGlobalContainerStack()
+        chamber_fan_nr = curaApp.getProperty("build_volume_fan_nr", "value")
+        extruder_count = curaApp.getProperty("machine_extruder_count", "value")
+        if heated_build_volume:
+            chamber_temp = curaApp.getProperty("build_volume_temperature", "value")
+            self._instance.setProperty("has_build_volume_heater", "value", heated_build_volume)
+            self._instance.setProperty("build_volume_temp", "value", chamber_temp)
+        try:
+            if chamber_fan_nr > 0:
+                self._instance.setProperty("enable_chamber_fan_setting", "value", True)
+        except:
+            pass
+
+    def getSettingDataString(self):
+        return """{
+            "name": "Annealing CoolDown or Filament Drying",
+            "key": "AnnealingOrDrying",
+            "metadata": {},
+            "version": 2,
+            "settings":
+            {
+                "enable_annealing":
+                {
+                    "label": "Enable the Script",
+                    "description": "If it isn't enabled it doesn't run.",
+                    "type": "bool",
+                    "default_value": true,
+                    "enabled": true
+                },
+                "cycle_type":
+                {
+                    "label": "Anneal Print or Dry Filament",
+                    "description": "Whether to Anneal the Print (by keeping the bed hot for a period of time), or to use the bed as a Filament Dryer.  If drying; you will still need to slice a model, but it will not print. The gcode will consist only of a short script to heat the bed, wait for a while, then turn the bed off.  The 'Z' will move to the max height and XY park position so the filament can be covered. The 'Hold Time', 'Bed Start Temp' and (if applicable) the 'Chamber Temp' come from these settings rather than from the Cura settings.  When annealing; the Timed Cooldown will commence when the print ends.",
+                    "type": "enum",
+                    "options":
+                    {
+
+                        "anneal_cycle": "Anneal Print",
+                        "dry_cycle": "Dry Filament"},
```

Comment:
```suggestion
                    {
                        "anneal_cycle": "Anneal Print",
                        "dry_cycle": "Dry Filament"
                     },
```

Some code cleaning remains :)